### DISCUSSION TITLE
caution-spring-application-properties-default-value/

### DISCUSSION THREAD

**Discussion by giscus** [Hyeon9mak/Hyeon9mak.github.io#179] [Blog comments]
# caution-spring-application-properties-default-value/

간만에 spring application properties 덕분에 고생 좀 했다. 정말 뼈저리게 고생했으니, 앞으로 까먹을 일은 없겠다.  

https://hyeon9mak.github.io/caution-spring-application-properties-default-value/

**Comment by hyunsu15** [comment]
현구막님 안녕하세요.  현구막님 블로그를 자주 보는 독자입니다.

아시는 내용일 수 있으나, 제가 쓰는 방식도 공유해볼까 싶어서 남깁니다!

저 같은 경우는 application-환경.yml 이런식으로 작성하여 환경을 분리합니다.

여기서 환경이라는 작성한 곳에 profile 이름을 넣어주면 됩니다. 
(ex.application-dev.yml 라고 작성하면 dev라는 profile, application-localtest.yml라고 작성하면 localtest라는 profile)

이렇게 하면 장점은 프로파일마다 환경이 명확히 분리가 됩니다.

단점이라면, 현구막님 사례처럼 사용하고 싶을때,  아쉬울 때가 있습니다.

저는 혼란을 방지하기 위해 귀찮더라도, 이 방법을 사용하고 있습니다.

항상 좋은 글 공유해주셔서 감사합니다. 다음 글도 기대하고 있겠습니다.

**Reply by Hyeon9mak** [reply]
안녕하세요 현수님! 좋은 방법 공유해주셔서 감사합니다.
다 같이 일하는 환경에서는 편의만큼이나 혼란방지가 중요하다는 걸 배웠습니다.

'매 환경마다 중복된 값을 기입할 필요 없이 하나의 설정으로 다룰 수 있다.' 라는 메리트 때문에
하나의 yml 파일에 환경 별로 구역을 나누어 다루는 방법을 선호하고 있었는데,
글 내용과 같은 사고를 겪고 나니 현수님께서 제안해주신 방법의 메리트가 특히 강하게 느껴집니다 👍
팀 회고 시간에 제안해주신 방법으로 이야기를 꺼내볼까 합니다...!

글 좋게 봐주셔서 감사합니다! 🙇‍♂️
